### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by tokejepsen**
**Goal**

Implement #16

**Implementation**

Currently you need to install the ```pymongo``` module with ```pip``` to run the tests, because testing happens through Linux with the Docker Toolbox terminal.

### CODE REVIEW

**Comment by tokejepsen** [comment]
Sure, we could do that. I'm not familiar with nose, but will have a look at the other repositories.

**Comment by mottosso** [comment]
It looks like the existing test can look the same, since it doesn't use anything pytest specific. Just replace `pytest.main` with [`nose.main`](https://github.com/pyblish/pyblish-base/blob/master/run_testsuite.py#L18).

**Comment by tokejepsen** [comment]
This is ready for a merge, unless we want Travis in on this as well?

**Comment by mottosso** [comment]
Travis would be good.

**Comment by tokejepsen** [comment]
Could you enable Travis for this repository?

**Comment by tokejepsen** [comment]
Are you able to look at the failed CI?

Dont know Travis well enough to know where to go with the error logs.

File: tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import os
+import subprocess
+
+
+def test_backup_restore():
+    """Backup and restore works"""
+
+    subprocess.call(
+        [
+            "docker", "run", "-d", "--rm",
```
Comment (mottosso) [inline tests/test_cli.py]: I think you can call this prior to running the tests, in the .travis file for example. That way you can debug Docker separately from the tests themselves.

**Comment by mottosso** [comment]
Looks like it's the `--import` command that fails. Could also be docker being unable to bind to those ports.

File: tests/test_cli.py (hunk 1 above)
Comment (tokejepsen) [reply tests/test_cli.py]: I'll need to kill the container to get a new database for restore. This is why I didn't want to assume the container is running.

**Comment by tokejepsen** [comment]
Is it supposed to display the help information for the ```docker``` command?

File: tests/test_cli.py (hunk 1 above)
Comment (mottosso) [reply tests/test_cli.py]: Probably better to wipe it instead.

**Comment by mottosso** [comment]
No, it looks like the command is mistyped.

**Comment by mottosso** [comment]
PyQt5 for running the tests? If it's a dependency, it's probably just an unnecessary import someplace.

**Comment by tokejepsen** [comment]
> PyQt5 for running the tests? If it's a dependency, it's probably just an unnecessary import someplace.

Its being checked for [here](https://github.com/getavalon/docker/blob/master/volume/avalon_cli.py#L84). Think we spoke about maybe including PyQt5 for Linux as well.

**Comment by mottosso** [comment]
> Think we spoke about maybe including PyQt5 for Linux as well.

Yeah, we did and we should, just wasn't sure you wanted to do bundling in the same PR as tests. I think bundling may need some work; for example, we may need to compile things ourselves to avoid dependencies of a given distribution, like CentOS or Ubuntu etc.

**Comment by tokejepsen** [comment]
> Yeah, we did and we should, just wasn't sure you wanted to do bundling in the same PR as tests. 

Yeah, I did give it a quick try, but gave up so just installing PyQt5 with pip and Travis instead.

**Comment by mottosso** [comment]
> Starting avalon-launcher

Looks like it isn't seeing the arguments to avalon, and are running it without, which would boot up the Launcher, which requires a screen. We could add the Xvfb fake screen to the tests, but really, we shouldn't require PyQt for these tests. Should we just remove the check when not running the launcher?

**Comment by tokejepsen** [comment]
> Looks like it isn't seeing the arguments to avalon

Yeah, I dont really get why that would be. Only thing I could imagine is something to do with the execution mode perhaps.

> Should we just remove the check when not running the launcher?

Yeah, I think that might be the best option.

**Comment by tokejepsen** [comment]
I dont know how to proceed with this. I'm not developing on Linux, so it makes it very tricky for me to troubleshoot issues on Travis.

I would suggest instead that we setup CI on Appveyor, where I'm more familar and the testing can happen there for the time being, until someone working on Linux can step in a fix the Travis issues.

**Comment by tokejepsen** [comment]
Closing this and opening a new with multi container support.

**Pull request closed without merging**